### PR TITLE
Minor additions and important bug fix in ImGui.GetIO()

### PR DESCRIPTION
**Pull request by MasterQ32**
I added some useful functions, also fixed a Bug in `ImGui.GetIO()`:

When using multiple contexts, the IO structure changes per context and must not be cached in order to adapt to work with `ImGui::SetContext`

### CODE REVIEW

File: src/ImGui.NET/ImGui.cs [hunk 1]
```diff
@@ -122,7 +133,7 @@ public static bool InvisibleButton(string id, Vector2 size)
         {
             return ImGuiNative.igInvisibleButton(id, size);
         }
-
+
```
Comment (mellinoe) [inline src/ImGui.NET/ImGui.cs]: nit: unnecessary whitespace change

File: src/ImGui.NET/ImGui.cs [hunk 2]
```diff
@@ -38,6 +39,11 @@ public static void PushID(string id)
             ImGuiNative.igPushIDStr(id);
         }
 
+        public static void PushID(object id)
```
Comment (mellinoe) [inline src/ImGui.NET/ImGui.cs]: Seems okay, although I think realistically you could just call GetHashCode yourself...

File: src/ImGui.NET/IO.cs [hunk 3]
```diff
@@ -265,6 +265,12 @@ public Font AddDefaultFont()
             return new Font(nativeFontPtr);
         }
 
+        public Font AddDefaultFont(FontConfig cfg)
```
Comment (mellinoe) [inline src/ImGui.NET/IO.cs]: Seems like a fine addition.

File: src/ImGui.NET/ImGui.cs [hunk 4]
```diff
@@ -22,9 +22,10 @@ public static void Shutdown()
             ImGuiNative.igShutdown();
         }
 
-        private static unsafe readonly IO s_io = new IO(ImGuiNative.igGetIO());
+        //private static unsafe readonly IO s_io = new IO(ImGuiNative.igGetIO());
```
Comment (mellinoe) [inline src/ImGui.NET/ImGui.cs]: Can you delete this line instead of commenting it out?

File: src/ImGui.NET/ImGui.cs [hunk 5]
```diff
@@ -22,9 +22,10 @@ public static void Shutdown()
...
 
-        public static unsafe IO GetIO() => s_io;
+        //public static unsafe IO GetIO() => s_io;
```
Comment (mellinoe) [inline src/ImGui.NET/ImGui.cs]: Ditto -- delete

File: src/ImGui.NET/ImGui.cs (hunk 2 above)
Comment (MasterQ32) [reply src/ImGui.NET/ImGui.cs]: > although I think realistically you could just call GetHashCode yourself
I think i'm not the only one using this technique for assigning unique IDs to same-name windows/controls, also this would replicate the `ImGui::PushID(void * id)` behaviour of C++ API

**Comment by MasterQ32** [comment]
I messed up in the github online editor, wanted to remove those two comments. Will fix it as soon as i get my hands on the dev PC again

**Comment by mellinoe** [comment]
@MasterQ32 I fixed the `GetIO()` bug separately (as well as `GetStyle()`), so you'll need to rebase that stuff out of this PR.

**Comment by mellinoe** [comment]
Please rebase and re-open if this is still important.
